server/room-battle.ts

```diff
@@ -1111,6 +1114,16 @@ export class RoomBattle extends RoomGames.RoomGame {
 		// @ts-ignore
 		this.room = null;
 	}
+	onChatMessage(message: string, user: User) {
+		void this.stream.write(`>chat ${user.getIdentity(this.room.roomid)}|${message}`);
+		return ' ';
+	}
+	getLog(): Promise<string[]> {
+		void this.stream.write('>requestlog');
+		return this.stream.read().then(() => {
```

Comment:
This is super brittle and a race condition just waiting to happen.

Reply:
I'm actually surprised this works at all, this should nearly definitely result in data loss: You're reading from the stream and discarding its data.